holoviews/plotting/bokeh/raster.py

```diff
@@ -130,26 +136,31 @@ class HeatmapPlot(ColorbarPlot):
     def _axes_props(self, plots, subplots, element, ranges):
         dims = element.dimensions()
         labels = self._get_axis_labels(dims)
-        xvals, yvals = [element.dimension_values(i, False)
+        agg = element.gridded
+        xvals, yvals = [unique_array(agg.dimension_values(i, False))
```

Comment:
I thought gridded ``Datasets`` have the 1D coordinate arrays available. Is the uniqueness being applied over the 2D set of samples or the 1D sequence?